lib/fli/FliImpl.cpp

```diff
@@ -107,11 +107,179 @@ void FliImpl::sim_end(void)
     mti_Cmd(stop);
 }
 
+bool FliImpl::isValueConst(int kind)
+{
+    return (kind == accGeneric || kind == accVHDLConstant);
+}
+
+bool FliImpl::isValueLogic(mtiTypeIdT type)
+{
+    mtiInt32T numEnums = mti_TickLength(type);
+    if (numEnums == 2) {
+        char **enum_values = mti_GetEnumValues(type);
+        std::string str0 = enum_values[0];
+        std::string str1  = enum_values[1];
+
+        if (str0.compare("'0'") == 0 && str1.compare("'1'") == 0) {
+            return true;
+        }
+    } else if (numEnums == 9) {
+        const char enums[9][4] = {"'U'","'X'","'0'","'1'","'Z'","'W'","'L'","'H'","'-'"};
+        char **enum_values = mti_GetEnumValues(type);
+
+        for (int i = 0; i < 9; i++) {
+            std::string str = enum_values[i];
+            if (str.compare(enums[i]) != 0) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    return false;
+}
+
+bool FliImpl::isValueChar(mtiTypeIdT type)
+{
+    return (mti_TickLength(type) == 256);
```

Comment:
@stuarthodgson There is no define, this is just how many enum values are defined for a type CHARACTER in the VHDL LRM.  I will add a static const int to the function to give the a more descriptive meaning.